libs/feature/catalog/src/lib/organisations/service/organisations-from-groups.service.ts

```diff
@@ -0,0 +1,75 @@
+import { Injectable } from '@angular/core'
+import { GroupApiModel, SearchApiService } from '@geonetwork-ui/data-access/gn4'
+import { ElasticsearchService, Organisation } from '@geonetwork-ui/util/shared'
+import { combineLatest } from 'rxjs'
+import { map, shareReplay } from 'rxjs/operators'
+import { GroupService } from '../../group/group.service'
+import { OrganisationsServiceInterface } from './organisations.service.interface'
+import { TranslateService } from '@ngx-translate/core'
+import { LANG_2_TO_3_MAPPER } from '@geonetwork-ui/util/i18n'
+
+const IMAGE_URL = '/geonetwork/images/harvesting/'
+
+type ESBucket = {
+  key: string
+  doc_count: number
+}
+interface OrganisationAggsBucket extends ESBucket {
+  mail: {
+    buckets: ESBucket[]
+  }
+}
+@Injectable()
+export class OrganisationsFromGroupsService
+  implements OrganisationsServiceInterface
+{
+  organisations$ = combineLatest([
+    this.fetchGroupsAggregation(),
+    this.groupService.groups$,
+  ]).pipe(
+    map(([groupsAgg, groups]) => {
+      return this.mapGroups(groupsAgg, groups)
+    }),
+    shareReplay()
+  )
+  organisationsCount$ = this.organisations$.pipe(map((orgs) => orgs.length))
+
+  constructor(
+    private esService: ElasticsearchService,
+    private searchApiService: SearchApiService,
+    private groupService: GroupService,
+    private translateService: TranslateService
+  ) {}
+
+  private fetchGroupsAggregation() {
+    return this.searchApiService
+      .search(
+        'bucket',
+        JSON.stringify(
+          this.esService.getSearchRequestBody({
+            groups: {
+              terms: {
+                field: 'groupOwner',
+                size: 5000,
+              },
+            },
+          })
+        )
+      )
+      .pipe(map((response) => response.aggregations.groups.buckets))
+  }
+
+  private mapGroups(groupBuckets: any[], groups: GroupApiModel[]) {
```

Comment:
Adding the return type here helps for clarity and quickly understanding what does this function do.